File: csharp/high-scores/HighScores.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class HighScores
{
    private readonly List<int> scores;

    public HighScores(List<int> list) =>
        scores = list ?? new List<int>();

    public List<int> Scores() => scores;

    public int Latest() => scores.LastOrDefault();

    public int PersonalBest() => scores.Max();

    public List<int> PersonalTopThree() => 
        scores.OrderByDescending(x => x).Take(3).ToList();
}

File: javascript/high-scores/high-scores.js
export class HighScores {
  constructor(input) {
    this._scores = input;
  }

  get scores() {
    return this._scores;
  }

  get latest() {
    return this._scores[this._scores.length - 1];
  }

  get personalBest() {
    return Math.max(...this._scores);
  }

  get personalTopThree() {
    return this._scores.sort((a, b) => b - a).slice(0, 3);
  }
}
